lib/chore/queues/sqs/consumer.rb

```diff
@@ -88,9 +88,9 @@ def delay(item, backoff_calc)
         def handle_messages(&block)
           msg = queue.receive_messages(:max_number_of_messages => sqs_polling_amount, :attribute_names => ['ApproximateReceiveCount'])
           messages = *msg
-
+          received_timestamp = Time.now
           messages.each do |message|
-            unless duplicate_message?(message.message_id, message.queue_url, queue_timeout)
+            unless duplicate_message?(message.message_id, message.queue_url, queue_timeout, received_timestamp)
```

Comment:
For the **`Consumer`** dedupe check
The **`Consumer`** version passes `received_timestamp` straight into the duplicate check.

Reply:
This was actually what I had for a first draft for this PR. That first draft version changed some function definitions and needed additional tests. I ended up rewriting it to this current version which would be easier to test/review. I don't mind going back to it